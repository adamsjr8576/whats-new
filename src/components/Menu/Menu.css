nav {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 25%;
  height: 350px;
}

.topic-button {
  display: flex;
  align-items: center;
  width: 80%;
  min-width: 175px;
  padding: 10px;
  font-size: 135%;
  border: none;
  background: none;
  color: black;
  outline: 0;
}

.topic-button:hover {
  background-color: #DADADC;
  border-bottom-right-radius: 50px;
  border-top-right-radius: 50px;
  cursor: pointer;
}

.topic-button:active {
  box-shadow: -3px 3px grey;
  transform: translateX(3px);
}

.button-icon {
  height: 40px;
  width: 40px;
  margin-right: 20px;
}
