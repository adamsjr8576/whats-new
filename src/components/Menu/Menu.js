import React from 'react';
import './Menu.css';
import PropTypes from 'prop-types';

import entertainmentIcon from '../../images/entertainment.png';
import healthIcon from '../../images/health.png';
import scienceIcon from '../../images/science.png';
import localIcon from '../../images/local.png';
import technologyIcon from '../../images/technology.png';

const Menu = ({ changeTopic, loading }) => {
  return (
    <nav>
      <button className="topic-button local"
        type="button"
        disabled={loading}
        onClick={() => changeTopic('local')}>
        <img className="button-icon" src={localIcon} alt="map pin icon" /> Local
      </button>
      <button className="topic-button entertainment"
        type="button"
        disabled={loading}
        onClick={() => changeTopic('entertainment')}>
        <img className="button-icon" src={entertainmentIcon} alt="map pin icon" /> Entertainment
      </button>
      <button className="topic-button health"
        type="button"
        disabled={loading}
        onClick={() => changeTopic('health')}>
        <img className="button-icon" src={healthIcon} alt="map pin icon" /> Health
      </button>
      <button className="topic-button science"
        type="button"
        disabled={loading}
        onClick={() => changeTopic('science')}>
        <img className="button-icon" src={scienceIcon} alt="map pin icon" /> Science
      </button>
      <button className="topic-button technology"
        type="button"
        disabled={loading}
        onClick={() => changeTopic('technology')}>
        <img className="button-icon" src={technologyIcon} alt="map pin icon" /> Technology
      </button>
    </nav>
  );
}

Menu.propTypes = {
  changeTopic: PropTypes.func,
  loading: PropTypes.bool
}

export default Menu;
