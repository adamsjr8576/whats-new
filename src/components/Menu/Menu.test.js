import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Menu
      changeTopic={jest.fn()}
      loading={false}
      />);

      expect(wrapper).toMatchSnapshot()
  });

  it('should call changeTopic method on click', () => {
    const changeTopicMock = jest.fn();
    const wrapper = shallow(<Menu
      changeTopic={changeTopicMock}
      loading={false}
      />);

      wrapper.find('.local').simulate('click');
      wrapper.find('.entertainment').simulate('click');
      wrapper.find('.health').simulate('click');
      wrapper.find('.science').simulate('click');
      wrapper.find('.technology').simulate('click');

      expect(changeTopicMock).toHaveBeenCalledWith("local");
      expect(changeTopicMock).toHaveBeenCalledWith("entertainment");
      expect(changeTopicMock).toHaveBeenCalledWith("health");
      expect(changeTopicMock).toHaveBeenCalledWith("science");
      expect(changeTopicMock).toHaveBeenCalledWith("technology");
  });

});
