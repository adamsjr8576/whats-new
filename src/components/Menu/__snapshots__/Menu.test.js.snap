// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all data passed in correctly 1`] = `
<nav>
  <button
    className="topic-button local"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      alt="map pin icon"
      className="button-icon"
      src="local.png"
    />
     Local
  </button>
  <button
    className="topic-button entertainment"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      alt="map pin icon"
      className="button-icon"
      src="entertainment.png"
    />
     Entertainment
  </button>
  <button
    className="topic-button health"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      alt="map pin icon"
      className="button-icon"
      src="health.png"
    />
     Health
  </button>
  <button
    className="topic-button science"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      alt="map pin icon"
      className="button-icon"
      src="science.png"
    />
     Science
  </button>
  <button
    className="topic-button technology"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      alt="map pin icon"
      className="button-icon"
      src="technology.png"
    />
     Technology
  </button>
</nav>
`;
